sys/mips/include/setjmp.h

```diff
@@ -49,17 +54,27 @@
 #ifndef _LOCORE
 #ifndef __ASSEMBLER__
 #if __BSD_VISIBLE || __POSIX_VISIBLE || __XSI_VISIBLE
-#ifdef __mips_n32
-typedef struct _sigjmp_buf { long long _sjb[_JBLEN + 1]; } sigjmp_buf[1];
+#ifdef __CHERI__
+typedef struct _sigjmp_buf {
+	union {
```

Comment:
I wonder if a union really makes sense given that we never access the contents from C.  I might just stick an __aligned(sizeof(void * __capability)) on the struct.

Reply:
I'm extremely skeptical that we can allow the compiler to make that decision and have working software.